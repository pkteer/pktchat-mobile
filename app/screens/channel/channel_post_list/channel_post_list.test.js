// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import Preferences from '@mm-redux/constants/preferences';

import ChannelPostList from './channel_post_list';

describe('ChannelPostList', () => {
    const baseProps = {
        actions: {
            loadPostsIfNecessaryWithRetry: jest.fn(),
            getPostThread: jest.fn(),
            increasePostVisibility: jest.fn(),
            selectPost: jest.fn(),
        },
        channelId: 'channel-id',
        loadMorePostsVisible: false,
        refreshing: false,
        theme: Preferences.THEMES.denim,
        registerTypingAnimation: jest.fn(() => {
            return jest.fn();
        }),
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <ChannelPostList {...baseProps}/>,
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should add/remove typing animation on mount/unmount', () => {
        const wrapper = shallow(
            <ChannelPostList {...baseProps}/>,
        );
        const instance = wrapper.instance();

        expect(baseProps.registerTypingAnimation).toHaveBeenCalledTimes(1);
        expect(instance.removeTypingAnimation).not.toHaveBeenCalled();

        wrapper.unmount();
        expect(instance.removeTypingAnimation).toHaveBeenCalledTimes(1);
    });
});
