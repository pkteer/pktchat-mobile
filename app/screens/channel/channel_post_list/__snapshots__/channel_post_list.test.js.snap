// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelPostList should match snapshot 1`] = `
<ForwardRef(AnimatedComponentWrapper)
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(63,67,80,0.2)",
        "height": 1,
      }
    }
  />
  <Connect(InjectIntl(PostList))
    channelId="channel-id"
    extraData={false}
    indicateNewMessages={true}
    loadMorePostsVisible={false}
    location="Channel"
    onLoadMoreUp={[Function]}
    postIds={Array []}
    refreshing={false}
    renderFooter={[Function]}
    scrollViewNativeID="channel-id"
    showMoreMessagesButton={true}
    testID="channel.post_list"
  />
  <Connect(RetryBarIndicator) />
</ForwardRef(AnimatedComponentWrapper)>
`;
